docs/DeveloperGuide.md

```diff
@@ -453,6 +453,126 @@ For example:
 
 The `loadTasksFromFile` method reads the tasks from the file and populates the `TaskManager` with the loaded tasks.
 
+## Exceptions and Logging
+
+### Exceptions
+
+![Exceptions.png](images%2Fclass%2FExceptions.png)
+
+The project utilizes centralized exception handling through the `TaskManagerException` class and its subclasses, 
+`MarkTaskException` and `SetPriorityException`.
+
+The `TaskManagerException` class serves as the base exception class, providing common functionality for handling 
+exceptions related to the Task Manager. It includes methods to check the validity of dates, times, and task lists, 
+and throws appropriate exceptions with corresponding error messages.
+
+```java
+public class TaskManagerException extends Exception {
+    public static final String NOT_CURRENT_WEEK_MESSAGE = "The date must be within the current week. Please try again.";
+    public static final String NOT_CURRENT_MONTH_MESSAGE = "The date must be within the current month. " +
+            "Please try again.";
+    // Other error message constants
+
+    public TaskManagerException(String errorMessage) {
+        super(errorMessage);
+    }
+
+    public static void checkIfDateInCurrentWeek(LocalDate date, WeekView weekView) throws TaskManagerException {
+        // Implementation to check if date is in current week
+    }
+
+    // Other utility methods to check date, time, and task list validity
+}
+```
```

Comment:
Might be better to put in pseudocode instead in case the code ever changes. But since this is the final version of the code, it should be fine to leave it in!